d2common/d2data/d2datadict/objects.go

```diff
@@ -344,9 +344,10 @@ var Objects map[int]*ObjectRecord
 func LoadObjects(file []byte) {
 	Objects = make(map[int]*ObjectRecord)
 	data := strings.Split(string(file), "\r\n")[1:]
+	for idx := range data {
+		line := data[idx]
```

Comment:
Same comment for all similar modifications, but are you sure this is changing the compiled executable? Seems exactly like the kind of optimization I expect the compiler to do.

Reply:
> Same comment for all similar modifications, but are you sure this is changing the compiled executable? Seems exactly like the kind of optimization I expect the compiler to do.

We've seen it actually create copies when doing the two-value for range loops. So I prefer we just use indexes and not worry about it.

Reply:
The spec seem to say it shouldn't 
https://golang.org/ref/spec "For statements with range clause"
```
Range expression                          1st value          2nd value

array or slice  a  [n]E, *[n]E, or []E    index    i  int    a[i]       E
string          s  string type            index    i  int    see below  rune
map             m  map[K]V                key      k  K      m[k]       V
channel         c  chan E, <-chan E       element  e  E
```
It says the second value is literally generated using the `data[idx]` pattern

Reply:
https://play.golang.org/p/5LG71jC5tTj

Reply:
Yep, I've tested it, you can check it here: https://play.golang.org/p/2GMYyC3Ai7b

Reply:
I was wrong. This is incorrect. when you say `line := data[idx]` you have created a copy. The correct syntax would be `line := &data[idx]`